UI/ScottPlot/DataPersistedSample/ScottPlotDataPersistedSample/MainPage.xaml.cs

```diff
@@ -0,0 +1,324 @@
+    private List<Series> _cachedSeries = new(); // Cache the series data in memory
+    private const int _batchSize = 5; // For lazy loading: number of series to load at once
+    private int _currentBatchIndex = 0; // Track the current loading index for lazy loading
+    private DispatcherQueue _dispatcherQueue;
+    private readonly object _seriesLock = new();  // Object for locking the _cachedSeries list
+
+    private DateTime _lastRefreshTime = DateTime.MinValue; // For debounce
+
+    public MainPage()
+    {
+        this.InitializeComponent();
+        this.Loaded += MainPage_Loaded;
+        // Get the UI dispatcher
+        _dispatcherQueue = DispatcherQueue.GetForCurrentThread();
+    }
+
+    // Immediate feedback during initialization
+    private async void MainPage_Loaded(object sender, RoutedEventArgs e)
+    {
+        // Provide feedback while loading inital data
+        await UpdateUIStatusAsync("Loading initial data, please wait...");
+        try
+        {
+            // Initialize the database and wait for the filesystem to be ready
+            await InitializeDatabaseAsync();
+
+            // Load last used plot type
+            currentPlotType = _dbHelper.GetLastUsedPlotType() ?? currentPlotType;
+
+            // Start lazy loading of data
+            await LoadNextDataBatchAsync();  // Load the first batch of data
+            _isInitialized = true;  // Mark initialization as complete
+
+            // Initialize plot and update the status text with correct information
+            await InitializePlotAsync();
+
+            // Update the status text after data is loaded
+            UpdateStatusText(_cachedSeries.Sum(s => s.DataPoints.Length));
+        }
+        catch (Exception ex)
+        {
+            await UpdateUIStatusAsync($"Initialization failed: {ex.Message}");
+        }
+    }
+
+    private async Task InitializeDatabaseAsync()
+    {
+        _dbHelper = new DataService();
+        // Ensure async initialization for WebAssembly compatibility as well
+        await _dbHelper.InitializeAsync();
+    }
+
+    // Lazy load the next batch of data
+    private async Task LoadNextDataBatchAsync()
+    {
+        var nextBatch = await _dbHelper.GetSeriesBatchAsync(_currentBatchIndex, _batchSize);
+        lock (_seriesLock)
+        {
+            _cachedSeries.AddRange(nextBatch);
+        }
+        _currentBatchIndex += nextBatch.Count;
+
+        // Provide feedback after loading each batch
+        if (nextBatch.Count > 0)
+        {
+            await UpdateUIStatusAsync($"Loaded {_cachedSeries.Count} series so far...");
+        }
+        else
+        {
+            await UpdateUIStatusAsync($"All series loaded. Total: {_cachedSeries.Count} series.");
+        }
+
+        // Trigger further lazy loading if more data exists
+        if (nextBatch.Count == _batchSize)
+        {
+            await Task.Run(async () => await LoadNextDataBatchAsync());
+        }
```

Comment:
instead of chain calling, you can switch to `do ... while (nextBatch.Count != _batchSize)`